pkg/storage/storage.go

```diff
@@ -80,69 +97,124 @@ func (p *storage) GetContainerMetrics(pods ...apitypes.NamespacedName) ([]api.Ti
 
 	timestamps := make([]api.TimeInfo, len(pods))
 	resMetrics := make([][]metrics.ContainerMetrics, len(pods))
-
-	for i, pod := range pods {
-		metricPoint, present := p.pods[pod]
+	for podIdx, pod := range pods {
+		contPoints, present := p.pods[pod]
 		if !present {
 			continue
 		}
 
-		contMetrics := make([]metrics.ContainerMetrics, len(metricPoint.Containers))
-		var earliestTS *time.Time
-		for i, contPoint := range metricPoint.Containers {
-			contMetrics[i] = metrics.ContainerMetrics{
+		var (
+			contMetrics = make([]metrics.ContainerMetrics, len(contPoints))
+			earliestTS  time.Time
+			window      time.Duration
+			contIdx     int
+		)
+		for contName, contPoint := range contPoints {
+			prevContPoint := p.prevPods[pod][contName]
+			if prevContPoint.Timestamp.IsZero() {
+				continue
+			}
```

Comment:
To make the code more clearer, could we check if map has key, instead of checking if response is non empty.
```suggestion
			prevContPoint, found := p.prevPods[pod][contName]
			if !found {
				continue
			}
```

Reply:
Good point, the fact that I was creating entries for unexistant metrics points made the checks quite unnatural as my only way to verify their presence was to check the timestamp value. I updated the storing part to verify that a previous point exist before adding the entry to the map which made the changes you requested possible.